pkg/minikube/reason/reason.go

```diff
@@ -183,120 +235,211 @@ var (
-	GuestNodeAdd                  = Kind{ID: "GUEST_NODE_ADD", ExitCode: ExGuestError}
-	GuestNodeDelete               = Kind{ID: "GUEST_NODE_DELETE", ExitCode: ExGuestError}
-	GuestNodeProvision            = Kind{ID: "GUEST_NODE_PROVISION", ExitCode: ExGuestError}
-	GuestNodeRetrieve             = Kind{ID: "GUEST_NODE_RETRIEVE", ExitCode: ExGuestNotFound}
-	GuestNodeStart                = Kind{ID: "GUEST_NODE_START", ExitCode: ExGuestError}
-	GuestPause                    = Kind{ID: "GUEST_PAUSE", ExitCode: ExGuestError}
-	GuestProfileDeletion          = Kind{ID: "GUEST_PROFILE_DELETION", ExitCode: ExGuestError}
-	GuestProvision                = Kind{ID: "GUEST_PROVISION", ExitCode: ExGuestError}
+	// minikube failed to load cached images
+	GuestCacheLoad = Kind{ID: "GUEST_CACHE_LOAD", ExitCode: ExGuestError}
+	// minikube failed to setup certificates
+	GuestCert = Kind{ID: "GUEST_CERT", ExitCode: ExGuestError}
+	// minikube failed to access the control plane
+	GuestCpConfig = Kind{ID: "GUEST_CP_CONFIG", ExitCode: ExGuestConfig}
+	// minikube failed to properly delete a resource, such as a profile
+	GuestDeletion = Kind{ID: "GUEST_DELETION", ExitCode: ExGuestError}
+	// minikube failed to list images on the machine
+	GuestImageList = Kind{ID: "GUEST_IMAGE_LIST", ExitCode: ExGuestError}
+	// minikube failed to pull or load an image
+	GuestImageLoad = Kind{ID: "GUEST_IMAGE_LOAD", ExitCode: ExGuestError}
+	// minikube failed to remove an image
+	GuestImageRemove = Kind{ID: "GUEST_IMAGE_REMOVE", ExitCode: ExGuestError}
+	// minikube failed to build an image
+	GuestImageBuild = Kind{ID: "GUEST_IMAGE_BUILD", ExitCode: ExGuestError}
+	// minikube failed to load host
+	GuestLoadHost = Kind{ID: "GUEST_LOAD_HOST", ExitCode: ExGuestError}
+	// minkube failed to create a mount
+	GuestMount = Kind{ID: "GUEST_MOUNT", ExitCode: ExGuestError}
+	// minkube failed to update a mount
+	GuestMountConflict = Kind{ID: "GUEST_MOUNT_CONFLICT", ExitCode: ExGuestConflict}
+	// minikube failed to add a node to the cluster
+	GuestNodeAdd = Kind{ID: "GUEST_NODE_ADD", ExitCode: ExGuestError}
+	// minikube failed to remove a node from the cluster
+	GuestNodeDelete = Kind{ID: "GUEST_NODE_DELETE", ExitCode: ExGuestError}
+	// minikube failed to provision a node
+	GuestNodeProvision = Kind{ID: "GUEST_NODE_PROVISION", ExitCode: ExGuestError}
+	// minikube failed to retrieve information for a cluster node
+	GuestNodeRetrieve = Kind{ID: "GUEST_NODE_RETRIEVE", ExitCode: ExGuestNotFound}
+	// minikube failed to startup a cluster node
+	GuestNodeStart = Kind{ID: "GUEST_NODE_START", ExitCode: ExGuestError}
+	// minikube failed to pause the cluster process
+	GuestPause = Kind{ID: "GUEST_PAUSE", ExitCode: ExGuestError}
+	// minikube failed to delete a machine profile directory
+	GuestProfileDeletion = Kind{ID: "GUEST_PROFILE_DELETION", ExitCode: ExGuestError}
+	// minikube failed while attempting to provision the  host
```

Comment:
I think this should be "provision the guest", @sharifelgamal right? The error message [here](https://github.com/andriyDev/minikube/blob/4f114dafb9289cf966b914d8d86e26c31918310a/cmd/minikube/cmd/start.go#L1516) is wrong I think, this should probably say "error provisioning guest". We should fix that in this PR.

Reply:
Hey @andriyDev, @sharifelgamal 
I've done updating the rest of comments mentioned by andriy, any update/help on this? Should I change it here to Guest, as well as in the location you mentioned?

Reply:
so yeah, in this case, guest and host are being used interchangeably. what it means is that we failed to start the VM or container that minikube lives in. I agree with @andriyDev in this case, change the wording here and in the error message itself.